drivers/charger/charger_max20335.c

```diff
@@ -389,23 +389,8 @@ static int max20335_set_prop(const struct device *dev, charger_prop_t prop,
 			     const union charger_propval *val)
 {
 	struct charger_max20335_data *data = dev->data;
-	int ret;
 
 	switch (prop) {
-	case CHARGER_PROP_CONSTANT_CHARGE_CURRENT_UA:
-		ret = max20335_set_constant_charge_current(dev, val->const_charge_current_ua);
-		if (ret == 0) {
-			data->charge_current_ua = val->const_charge_current_ua;
-		}
-
-		return ret;
-	case CHARGER_PROP_CONSTANT_CHARGE_VOLTAGE_UV:
-		ret =  max20335_set_constant_charge_voltage(dev, val->const_charge_voltage_uv);
-		if (ret == 0) {
-			data->charge_voltage_uv = val->const_charge_voltage_uv;
-		}
-
-		return ret;
```

Comment:
I disagree with this commit, I can think of a few reasons why an application may need to adjust these targets. 

https://home.jeita.or.jp/upload_file/20210407112651_JbDgUSWGiB.pdf
What if the charging task wants to adjust the current (CC) and voltage (CV) targets based on the thermistor (reducing CC or CV if it is too hot or cold to charge at full capacity, safely)?

What if the system knows the user is asleep and wants to charge at a lower CC overnight?

How about if you clamp the maximum settable CC and CV targets to what is described in devicetree and allow runtime configuration instead? Please let me know if this comment is clear enough.

Reply:
I think I got your point. Now I see two issues with this commit.

1) Order of commits is not correct. This one should be applied after https://github.com/zephyrproject-rtos/zephyr/pull/69910/commits/21537aeeab0a57391366303c16854db7cd19ae4b
2) Based on your input, `CHARGER_PROP_CONSTANT_CHARGE_VOLTAGE_UV` should stay.

If an application needs to set `CHARGER_PROP_CONSTANT_CHARGE_CURRENT_UA`, the property should be added again but different code needs to be called underneath (https://github.com/zephyrproject-rtos/zephyr/pull/69910/commits/21537aeeab0a57391366303c16854db7cd19ae4b again).
I propose to cover this within a separate PR.

Please confirm that we are on the same page.

Reply:
We are on the same page. 
I am okay with leaving reimplementing the `CHARGER_PROP_CONSTANT_CHARGE_CURRENT_UA` setter properly for a future PR. Given that the power stage is linear, the `CHARGER_PROP_CONSTANT_CHARGE_CURRENT_UA` is set by the input current limit indirectly so moving the value around at runtime could introduce new issues.

Reply:
Hi @rriveramcrus 
I have reworked discussed part. Can you re-review please ?